src/coreclr/jit/optimizer.cpp

```diff
@@ -7539,6 +7539,15 @@ bool Compiler::optComputeLoopSideEffectsOfBlock(BasicBlock* blk)
                         }
                         break;
 
+#ifdef FEATURE_HW_INTRINSICS
+                    case GT_HWINTRINSIC:
+                        if (tree->AsHWIntrinsic()->OperIsMemoryStore())
```

Comment:
Should we be specially tracking loads at all either?

Reply:
I don't think so. This method must account for all stores that could modify the heaps such that VN can then bypass creating a PHI for the memory at loop entry (enabling VN-ing of things not modified in the loop with VNs from before the loop was entered). So it only needs to know about stores.

> What about volatile loads?

Good point. The method does not account for them today, but in VN they are accounted for by modifying the heaps, so it would seem like it should. But it will be hard/very hard to create a test that fails if don't, I think.

Do we have explicitly volatile HWI/SIMD loads?

Reply:
We have the memory fences (`LoadFence`, `StoreFence`, `MemoryFence`) and the explicitly serializing instructions (`CPUID`), as well as a few other scenarios (including `non-temporal loads/stores`); but I don't think we have anything that counts as a "volatile load/store".

Reply:
We might actually not be modeling the fences correctly in `OperRequiresAsgFlag`, just from a quick search...

Reply:
Will resolve this as "it is how it is", we will need to understand more broadly how we want to treat these intrinsics before making changes.

Today I don't think we mutate the heap for them in VN, for example, we don't set `GTF_ORDER_SIDEEFF`, we don't check them in `gtDebugCheckFlags`, we will contain things across them, etc. Really, the only thing preventing liveness from removing them, it seems, is the fact that all `HWINTRINSIC` nodes are treated as non-removable.